mobile/lib/models/search/search_result.dart

```diff
@@ -15,5 +16,7 @@ abstract class SearchResult {
 
   List<EnteFile> resultFiles();
 
+  List<Collection> resultCollections();
```

Comment:
Let's not add this abstract method. All sub classes will have to implement this but it's not _fully_ relevant to any of the current subclasses. It's only relevant to `GenericSearchResult` and that too only when `type()` is `ResultType.shared`. Alternate options are:
1.  Store `resultCollections` in `params` of `GenericSearchResult`, or
2. Create a new sub class for better separation called `ContactSearchResult` that extends `SearchResult` and there you can have a `resultCollections` getter.

Easiest would be option 1 and a much cleaner code would be option 2 which would require some refactoring.